website_oca_integrator/models/res_partner.py

```diff
@@ -0,0 +1,281 @@
+                                 string='Authors', inverse_name='partner_id',
+                                 readonly=True)
+
+    developed_module_ids = fields.Many2many(
+        string='Developed Modules',
+        comodel_name='product.template',
+        relation='product_module_res_partner_rel',
+        column1='partner_id',
+        column2='product_template_id',
+        compute='_compute_developed_modules',
+        store=True)
+
+    favourite_module_ids = fields.Many2many(
+        string='Favourite Modules',
+        comodel_name='product.template',
+        relation='favourite_product_module_res_partner_rel',
+        column1='partner_id',
+        column2='product_template_id', readonly=True)
+
+    sponsorship_line_ids = fields.One2many(
+        string='Sponsorship Activities',
+        comodel_name='sponsorship.line',
+        inverse_name='partner_id')
+
+    contributor_module_line_ids = fields.One2many(
+        string='Contributed Modules',
+        comodel_name='contributor.module',
+        inverse_name='partner_id', readonly=True)
+
+    @api.multi
+    @api.depends('github_organization')
+    def _compute_github_organization_url(self):
+        github_url = 'https://github.com/'
+        for record in self:
+            if record.is_company:
+                github_organization = record.github_organization
+                if github_organization:
+                    record.github_organization_url = \
+                        github_url + github_organization
+                else:
+                    record.github_organization_url = False
+
+    @api.depends(
+        'child_ids',
+        'child_ids.membership_state',
+        'child_ids.parent_id')
+    def _compute_integrator(self):
+        for partner in self:
+            if any([(child.github_login or child.membership_state == 'paid')
+                    for child in partner.child_ids]):
+                partner.is_integrator = True
+            else:
+                partner.is_integrator = False
+
+    @api.depends('child_ids.github_login', 'child_ids.parent_id')
+    def _compute_contributor_count(self):
+        contributor_data = self.read_group(
+            domain=[('parent_id', 'in', self.ids),
+                    ('github_login', '!=', False)], fields=['parent_id'],
+            groupby=['parent_id'])
+
+        contributor_mapped_data = dict(
+            (item['parent_id'][0], item['parent_id_count']) for item in
+            contributor_data)
+
+        for partner in self:
+            partner.contributor_count = contributor_mapped_data.get(
+                partner.id, 0)
+
+    @api.depends('child_ids.membership_state', 'child_ids.parent_id')
+    def _compute_member_count(self):
+        member_data = self.read_group(
+            domain=[('parent_id', 'in', self.ids),
+                    ('membership_state', '=', 'paid')], fields=['parent_id'],
+            groupby=['parent_id'])
+
+        member_mapped_data = dict(
```

Comment:
Are you sure? What kind of loop is this?  Its not a list comprehension.  Anyway

`{item['parent_id'][0]: item['parent_id_count'] for item in member_data}`